### PR TITLE
Name and Recipe Change (Steel Knuckles to Scrap Knuckles)

### PR DESCRIPTION
**Pull request by slimeboy460**
#### Summary

SUMMARY: None

#### Purpose of change

Logic and general update, to improve immersion and consistency.

#### Describe the solution

Changing the name from steel knuckles to scrap knuckles sounds more logical, as it isn't a study bit of steel, it's a collection of scrap metal thrown together with a hammer. I added metalworking just as a side thing, if anyone thinks this prof doesn't work then I'm more than happy to change it.

Martial Art code adjustments for consistency of ID, it'd be strange if in the next update you couldn't use your steel knuckles with whatever martial arts you main.

#### Testing

JSON lint

### CODE REVIEW

File: data/json/items/melee/unarmed_weapons.json [hunk 1]
```diff
@@ -84,10 +84,10 @@
   },
   {
     "type": "GENERIC",
-    "id": "knuckle_steel",
+    "id": "knuckle_scrap",
```
Comment (Qrox) [inline data/json/items/melee/unarmed_weapons.json:87]: Please also add a migration entry to `data/json/items/migration.json` to convert `knuckle_steel` items in old saves to `knuckle_scrap`.

Comment (Qrox) [reply data/json/items/melee/unarmed_weapons.json:87]: There's also `"overlay_wielded_knuckle_steel"` in `data/mods/Chibi_Ultica/mod_tileset.json`.

Comment (slimeboy460) [reply data/json/items/melee/unarmed_weapons.json:87]: Oh, gotcha, thanks for notifying me of that. I’ll get right to it.

**Comment by actual-nh** [comment]
Looks like some test cases need altering to match.

**Comment by slimeboy460** [comment]
> Looks like some test cases need altering to match.

I've never ran into test cases, can you explain them a bit more?

File: data/json/items/migration.json [hunk 2]
```diff
@@ -1298,5 +1298,10 @@
     "id": "jug_plastic_sealed",
     "type": "MIGRATION",
     "replace": "jug_plastic"
+  },
+  {
+    "id": "knuckles_steel",
```
Comment (Qrox) [inline data/json/items/migration.json:1303]:
```suggestion
    "id": "knuckle_steel",
```

File: data/json/items/migration.json [hunk 3]
```diff
@@ -1298,5 +1298,10 @@
...
+    "type": "MIGRATION",
+    "replace": "knuckles_scrap"
```
Comment (Qrox) [inline data/json/items/migration.json:1305]:
```suggestion
    "replace": "knuckle_scrap"
```

File: data/json/recipes/weapon/bashing.json [hunk 4]
```diff
@@ -246,11 +246,12 @@
   {
     "type": "recipe",
     "activity_level": "LIGHT_EXERCISE",
-    "result": "knuckle_steel",
+    "result": "knuckle_scrap",
```
Comment (Qrox) [inline data/json/recipes/weapon/bashing.json:249]: The old recipe id needs to be added to `data/json/obsolete.json` to avoid error message in old saves.

**Comment by Qrox** [comment]
>> Looks like some test cases need altering to match.

> I've never ran into test cases, can you explain them a bit more?

`knuckle_steel` is also referenced in `tests/effective_dps_test.cpp`, fixing that should fix the test.

**Comment by actual-nh** [comment]
> > I've never ran into test cases, can you explain them a bit more?

Sorry, got distracted and forgot about the question...

> `knuckle_steel` is also referenced in `tests/effective_dps_test.cpp`, fixing that should fix the test.

It looks a bit weird because the test NPC is doing less damage without them, and that's being spotted. (The other errors are from knuckles_steel/knuckles_scrap.)

**Comment by slimeboy460** [comment]
It's still coming up with errors and I don't know how to read em, fuck

**Comment by Qrox** [comment]
It seems you didn't apply my suggested fixes above, which is why the test is still failing. (Though I did say "..., fixing that should fix the test." in the following comment, which might have misled you, so it's my bad.)

**Comment by actual-nh** [comment]
> It's still coming up with errors and I don't know how to read them, fuck

It's that they're being spelled places as "knuckles_steel" and "knuckles_scrap" instead of "knuckle_steel" and "knuckle_scrap". Qrox found some of them - accept the review changes and that should help. (Also note the need to put the old recipe id in data/json/obsolete.json that Qrox mentioned.)

**Comment by slimeboy460** [comment]
Ohhhh, okay, I missed those suggestions

File: data/json/obsolete.json [hunk 5]
```diff
@@ -1639,5 +1639,21 @@
...
+    "name": { "str": "pair of steel knuckles", "str_pl": "pairs of steel knuckles" },
+    "description": "A mass of scrap metal crudely beat into shape, with folded rags underneath to protect the wearers knuckles.  A good, quick weapon - but you have to get within punching range to use it.",
+    "category": "weapons",
+    "material": [ "steel" ],
+    "volume": "250 ml",
+    "weight": "430 g",
+    "bashing": 4,
+    "price_postapoc": 50,
+    "qualities": [ [ "HAMMER", 1 ] ],
+    "flags": [ "UNARMED_WEAPON" ]
```
Comment (Qrox) [inline data/json/obsolete.json:1657]: You should add the recipe instead of the item here.

Comment (actual-nh) [reply data/json/obsolete.json:1657]: Interesting. How does the game use it?

File: data/json/obsolete.json [hunk 6]
```diff
@@ -1639,5 +1639,20 @@
...
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_BASHING",
+    "skill_used": "fabrication",
+    "skills_required": [ "bashing", 1 ],
+    "proficiencies": [ { "proficiency": "prof_metalworking" } ],
+    "difficulty": 2,
+    "time": "30 m",
+    "autolearn": true,
+    "qualities": [ { "id": "HAMMER", "level": 1 }, { "id": "SAW_M", "level": 1 } ],
+    "components": [ [ [ "rag", 4 ] ], [ [ "scrap", 4 ] ] ]
```
Comment (Qrox) [inline data/json/obsolete.json:1656]: Actually, I just realized that this needs to be moved to `data/json/recipe/recipe_obsolote.json`, in the following form:
```json
{
  "type": "recipe",
  "result": "knuckle_steel",
  "obsolete": true
}
```
Sorry for the hassle.

Comment (slimeboy460) [reply data/json/obsolete.json:1656]: nah, it's fine. It helps me learn what to do in the future so it becomes less of a hassle overall. I'm assuming I should delete that file update

Comment (Qrox) [reply data/json/obsolete.json:1656]: Yeah, the changes to `data/json/obsolete.json` is not needed.

File: data/json/recipes/recipe_obsolete.json [hunk 7]
```diff
@@ -2680,5 +2680,10 @@
     "type": "recipe",
     "result": "foot_crank",
     "obsolete": true
+  },
+  {
+  "type": "recipe",
+  "result": "knuckle_steel",
+  "obsolete": true
```
Comment (Qrox) [inline data/json/recipes/recipe_obsolete.json:2687]:
```suggestion
    "type": "recipe",
    "result": "knuckle_steel",
    "obsolete": true
```
Json style.

File: data/json/obsolete.json (hunk 5 above)
Comment (Qrox) [reply data/json/obsolete.json]: Honestly I don't know, but old saves will show a debug message if the old recipe id is not obsoleted.

File: data/json/obsolete.json (hunk 6 above)
Comment (Qrox) [inline data/json/obsolete.json:1656]:
```suggestion
```

**Comment by slimeboy460** [comment]
so data/json/items/migration.json is conflicting, I'm gonna try and see if I can fix it

**Comment by slimeboy460** [comment]
I think you might've added it to locations for terrain. I'm not too sure about this though.

**Comment by slimeboy460** [comment]
Nevermind, disregard the previous comments.

File: data/json/items/migration.json [hunk 8]
```diff
@@ -1312,6 +1317,5 @@
   {
     "id": "rifle_3006",
     "type": "MIGRATION",
-    "replace": "remington_700"
```
Comment (Qrox) [inline data/json/items/migration.json:1315]: You accidentally removed a line when merging.

Comment (slimeboy460) [reply data/json/items/migration.json]: thanks for notifying me

**Comment by slimeboy460** [comment]
There’s still one failed thing, I tried looking over it but I can’t comprehend it yet.

**Comment by Qrox** [comment]
The failure seems unrelated to this PR.

**Comment by actual-nh** [comment]
Are you testing vs before or after [this commit](https://github.com/CleverRaven/Cataclysm-DDA/commit/ffd58c6f348aede16d2e7af7e4f5c58d0daf6a02)?

**Comment by slimeboy460** [comment]
I think this is based before that commit

**Comment by slimeboy460** [comment]
> The failure seems unrelated to this PR.


does that mean this is good to go?

**Comment by actual-nh** [comment]
Again, nothing to do with your PR - see #46256 for a related set of example unrelated PRs.

File: data/json/items/melee/unarmed_weapons.json (hunk 1 above)
Comment (Rivet-the-Zombie) [reply data/json/items/melee/unarmed_weapons.json:87]: Why not leave the item ids the same and just change the displayed name? Nothing breaks that way.

Comment (kevingranade) [reply data/json/items/melee/unarmed_weapons.json:87]: Seconded, don't change the ID, it's not displayed to the player anyway, so having it not match the name is not a big deal.

**Review by kevingranade — CHANGES_REQUESTED** [review]

**Comment by slimeboy460** [comment]
I was mainly changing it for people who wanted to upload a proper steel knuckle item. But seeing as that might be a while I'm fine with not updating it to scrap

**Comment by actual-nh** [comment]
> I was mainly changing it for people who wanted to upload a proper steel knuckle item. But seeing as that might be a while I'm fine with not updating it to scrap

Your logic makes sense to me (re if someone wanted to make a forged "steel knuckle"). But I suppose they could call it "knuckle_brass", since they're called "brass knuckles" despite not being made of brass...

**Comment by slimeboy460** [comment]
Done and dusted, let's see if it shall run

**Pull request merged**